table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,389 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Glue database implementation.
+ */
+public class GlueDatabase implements UnderDatabase {
+  private static final Logger LOG = LoggerFactory.getLogger(GlueDatabase.class);
+
+  private final UdbContext mUdbContext;
+  private final AWSGlueAsync mGlueClient;
+  private final UdbConfiguration mGlueConfiguration;
+  /** the name of the glue db. */
+  private final String mGlueDbName;
+
+  /** the owner name of glue database, which is an fake place holder. */
+  private final String mOwnerName = "PUBLIC_OWNER";
+  private final alluxio.grpc.table.PrincipalType mOwnerType = alluxio.grpc.table.PrincipalType.ROLE;
+
+  @VisibleForTesting
+  protected GlueDatabase(UdbContext udbContext, UdbConfiguration glueConfig, String glueDbName) {
+    mUdbContext = udbContext;
+    mGlueConfiguration = glueConfig;
+    mGlueClient = createAsyncGlueClient(glueConfig);
+    mGlueDbName = glueDbName;
+  }
+
+  /**
+   * Create an instance of the Glue database UDB.
+   *
+   * @param udbContext the db context
+   * @param configuration the configuration
+   * @return the new instance
+   */
+  public static GlueDatabase create(UdbContext udbContext, UdbConfiguration configuration) {
+    String glueDbName = udbContext.getUdbDbName();
+    if (glueDbName == null || glueDbName.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Glue database name cannot be empty: " + glueDbName);
+    } else if (configuration.get(Property.GLUE_REGION) == null) {
+      throw new IllegalArgumentException("GlueUdb Error: AWS region cannot be empty.");
+    }
+
+    return new GlueDatabase(udbContext, configuration, glueDbName);
+  }
+
+  @Override
+  public UdbContext getUdbContext() {
+    return mUdbContext;
+  }
+
+  @Override
+  public DatabaseInfo getDatabaseInfo() throws IOException {
+    LOG.info("Getting Glue database information from database: " + mGlueDbName + ".");
+    try {
+      GetDatabaseRequest dbRequest = new GetDatabaseRequest()
+          .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+          .withName(mGlueDbName);
+      GetDatabaseResult dbResult = mGlueClient.getDatabase(dbRequest);
+      Database glueDatabase = dbResult.getDatabase();
+      String glueDbLocation = glueDatabase.getLocationUri();
+      String glueDbDescription = glueDatabase.getDescription();
+      Map<String, String> glueParameters = new HashMap<>();
+      // Returned parameter can be null while Alluxio require non-null hash map for parameters
+      if (glueDatabase.getParameters() != null) {
+        glueParameters = glueDatabase.getParameters();
+      }
+      return new DatabaseInfo(
+          glueDbLocation,
+          mOwnerName,
+          mOwnerType,
+          glueDbDescription,
+          glueParameters);
+    } catch (EntityNotFoundException e) {
+      throw new IOException("Cannot find glue database" + mGlueDbName
```

Comment:
This needs a space after "database" and  space before the `e.getMessage()`. Also, can we add the catalog id to the message?

Reply:
Changed.